pkg/security/probe/dentry_resolver.go

```diff
@@ -268,10 +277,129 @@ func (dr *DentryResolver) ResolveFromMap(mountID uint32, inode uint64, pathID ui
 	return filename, err
 }
 
+func (dr *DentryResolver) preventSegmentMajorPageFault() {
+	// if we don't access the segment, the eBPF program can't write to it ... (major page fault)
+	dr.erpcSegment[0] = 0
+	dr.erpcSegment[os.Getpagesize()] = 0
+	dr.erpcSegment[2*os.Getpagesize()] = 0
+	dr.erpcSegment[3*os.Getpagesize()] = 0
+	dr.erpcSegment[4*os.Getpagesize()] = 0
+	dr.erpcSegment[5*os.Getpagesize()] = 0
+	dr.erpcSegment[6*os.Getpagesize()] = 0
+}
+
+// GetNameFromERPC resolves the name of the provided inode / mount id / path id
+func (dr *DentryResolver) GetNameFromERPC(mountID uint32, inode uint64, pathID uint32) (name string, err error) {
+	// create eRPC request
+	dr.erpcRequest.OP = ResolveSegmentOp
+	model.ByteOrder.PutUint64(dr.erpcRequest.Data[0:8], inode)
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[8:12], mountID)
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[12:16], pathID)
+	model.ByteOrder.PutUint64(dr.erpcRequest.Data[16:24], uint64(uintptr(unsafe.Pointer(&dr.erpcSegment[0]))))
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[24:28], uint32(dr.erpcSegmentSize))
+
+	// if we don't try to access the segment, the eBPF program can't write to it ... (major page fault)
+	dr.preventSegmentMajorPageFault()
+
+	if err = dr.erpc.Request(&dr.erpcRequest); err != nil {
+		return "", errors.Wrapf(err, "unable to get filename for mountID `%d` and inode `%d` with eRPC", mountID, inode)
+	}
+
+	seg := C.GoString((*C.char)(unsafe.Pointer(&dr.erpcSegment[16])))
+	if len(seg) == 0 || len(seg) > 0 && seg[0] == 0 {
+		return "", errors.Errorf("couldn't resolve segment (len: %d)", len(seg))
+	}
+	return seg, nil
+}
+
+// ResolveFromERPC resolves the path of the provided inode / mount id / path id
+func (dr *DentryResolver) ResolveFromERPC(mountID uint32, inode uint64, pathID uint32) (string, error) {
+	var filename, segment string
+	var err, resolutionErr error
+	var key PathKey
+	var val PathValue
+
+	// create eRPC request
+	dr.erpcRequest.OP = ResolvePathOp
+	model.ByteOrder.PutUint64(dr.erpcRequest.Data[0:8], inode)
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[8:12], mountID)
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[12:16], pathID)
+	model.ByteOrder.PutUint64(dr.erpcRequest.Data[16:24], uint64(uintptr(unsafe.Pointer(&dr.erpcSegment[0]))))
+	model.ByteOrder.PutUint32(dr.erpcRequest.Data[24:28], uint32(dr.erpcSegmentSize))
+
+	// if we don't try to access the segment, the eBPF program can't write to it ... (major page fault)
+	dr.preventSegmentMajorPageFault()
+
+	if err = dr.erpc.Request(&dr.erpcRequest); err != nil {
+		return "", err
+	}
+
+	var keys []PathKey
+	var segments []string
+
+	i := 0
+	for i < dr.erpcSegmentSize-17 {
```

Comment:
Is 17 sizeof(pathID) + 1 ?

Reply:
Yes, I added a comment :)

Reply:
Also I think it's supposed to be `<=` unless I forgot an edge case 🤔